src/Promise.php

```diff
@@ -243,7 +243,7 @@ private function invokeWaitFn()
         try {
             $wfn = $this->waitFn;
             $this->waitFn = null;
-            $wfn(true);
+            $wfn($this);
```

Comment:
Hi @mtdowling,

Thank you for reviewing. No pb, I just have to figure out how to make multiple PRs (I'm not a great pull-requester), and I'll update the docs.
I'll look at this next week.